core/modules/queue.cc

```diff
@@ -77,6 +78,13 @@ CommandResponse Queue::Init(const bess::pb::QueueArg &arg) {
     prefetch_ = true;
   }
 
+  if (arg.backpressure()) {
+    backpressure_ = true;
+    high_water_ = (uint64_t) (size_ * kHighWaterRatio);
```

Comment:
I would move this setting of high_water_, low_water_ and underload_ to a separate function, because it will need to be called not just here but also when size is set.